pkg/storage/reports/constraint_stats_report.go

```diff
@@ -197,9 +197,9 @@ func (r *replicationConstraintStatsReportSaver) loadPreviousVersion(
 
 		// if the row is nil then this is the first time we are running and the reload is needed.
 		if row != nil {
```

Comment:
probably `if len(row) > 0` here instead. Otherwise if you have a non-nil but empty slice, the case below will panic.